elastic/kibana
# [Infrastructure UI] Save selected tab for Hosts View into url state

## 📓 Summary

Closes #150856 

This PR refactor the existing implementation of the tabs, extracting this piece of logic into its own hook `useTabId` which stores the selected tab in the URL and allows sharing this preference with others.

To keep the optimization performed in [[Infrastructure UI] Add Alerts tab into Hosts View](https://github.com/elastic/kibana/pull/149579), has been necessary to push down the memoization of the AlertSummary component.

This PR also hide the count of alerts when it is equal to `0`.

## 🧪 Testing
- Navigate to the hosts' view
- Create an Inventory rule to generate alerts for all our hosts (it can be something like cpu-usage > 0%)
- Refresh the hosts' view page and start switching between tabs.
- Refresh again the page to verify the selected tab is preserved and its value is stored in the URL.

https://user-images.githubusercontent.com/34506779/220871592-8a4ae1b7-3b9c-4ab1-be3f-4bd673ab9f2f.mov
